js_compiler/compile.js

```diff
@@ -330,16 +426,97 @@ function Module() {
 
 function FirstPassCodeGen() {
 
+	this.globalContext = new GlobalContext();
+	this.functionContext = new FunctionContext(this.globalContext);
 	this.module = undefined;
 	this.func = undefined;
 
 	this.compile = function (syntax) {
 		this.module = new Module();
-		var func = new FunctionDefinition(null); // top level
+		var func = new FunctionDefinition(null, "<body>", 0); // top level
+		this.module.functions.push(func);
+		this.augmentAST(syntax);
+		this.functionContext.enterFunction(syntax.functionEntry);
 		this.handleBody(func, syntax.body);
+		func.instructions.push(new Instruction("END_SECTION", 0)); //end of top level function
```

Comment:
Can we insert a `(int-push-const 0) (return)` before the `end-section` marker?

Reply:
sure. that would make the <body> non-empty, and I strongly believe jitting that would not be a problem.